### PR TITLE
Updated read method used.

### PR DESCRIPTION
**Pull request by mbarth**
Jira: https://finhaven.atlassian.net/browse/P2P-1115

* reading the message of the TCP stream to its proper length

### CODE REVIEW

File: tests/work_queue.rs [hunk 1]
```diff
@@ -0,0 +1,67 @@
+extern crate beanstalkd;
```
Comment (evan-finhaven) [inline tests/work_queue.rs]: Let's rename the this file. Looks like you copied `work_queue.rs` over and didn't rename it ☺️

File: tests/work_queue.rs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        assert_eq!(result.is_ok(), true)
+    }
+}
+
+#[test]
+fn handle_large_message() {
+    let mut beanstalkd = Beanstalkd::localhost().unwrap();
+    beanstalkd.tube("large-file").unwrap();
+    let message = include_str!("../data/very-large-json-file.json");
+    let _ = beanstalkd.put(message, 0, 0, 10000);
```
Comment (evan-finhaven) [inline tests/work_queue.rs]: I'd include a comment in here about how large data triggered a bug, and that's why we have this test.

**Review by evan-finhaven — APPROVED** [review]
A couple small things. Good work! 🦋

File: tests/work_queue.rs (hunk 1 above)
Comment (mbarth) [reply tests/work_queue.rs]: Yes, good catch.

File: tests/work_queue.rs (hunk 2 above)
Comment (mbarth) [reply tests/work_queue.rs]: Right, will do.

**Pull request merged**